File: backend/app/models/backup_meta.py
"""
Backup Metadata Model
"""
from app import db
from app.models.base import BaseModel

class BackupMeta(BaseModel):
    """
    Metadata for encrypted backup files
    """
    __tablename__ = 'backups'

    filename = db.Column(db.String(255), nullable=False)
    file_size_bytes = db.Column(db.BigInteger, nullable=False)
    created_by_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    
    # User who created the backup
    created_by = db.relationship('User', backref='backups')

    def to_dict(self):
        data = super().to_dict()
        data.update({
            'filename': self.filename,
            'file_size_bytes': self.file_size_bytes,
            'created_by': self.created_by.display_name if self.created_by else 'System'
        })
        return data

File: hostelixpro/lib/models/backup_meta.dart
class BackupMeta {
  final int id;
  final String filename;
  final int fileSizeBytes;
  final int createdById;
  final String createdBy;
  final int createdAt;

  BackupMeta({
    required this.id,
    required this.filename,
    required this.fileSizeBytes,
    required this.createdById,
    required this.createdBy,
    required this.createdAt,
  });

  factory BackupMeta.fromJson(Map<String, dynamic> json) {
    return BackupMeta(
      id: json['id'],
      filename: json['filename'],
      fileSizeBytes: json['file_size_bytes'],
      createdById: json['created_by_id'] ?? 0,
      createdBy: json['created_by'] ?? 'System',
      createdAt: json['created_at'],
    );
  }
}
